### PR TITLE
[VL] Remove the registry for Velox's prestosql scalar functions

### PR DESCRIPTION
**Pull request by PHILO-HE**
## What changes were proposed in this pull request?

Currently, Gluten registers Velox's prestosql scalar functions. Thus, some of them can be used for Spark SQL if there is no semantic difference. However, there are some conflict issues, which makes unexpected function calling. See below links. We plan to register used presto scalar functions (see below list) in sparksql registry.

https://github.com/apache/incubator-gluten/discussions/4759
https://github.com/apache/incubator-gluten/pull/5150

Function list:
url_decode/url_encode
array_distinct
array_except
array_position
map_entries
map_keys
map_values
reverse
crc32
regexp_extract_all
like
abs (for long/short decimal)
asin
atan
cos
degrees

## How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by Yohahaha** [comment]
Hi, could you share the current status?

**Comment by PHILO-HE** [comment]
> Hi, could you share the current status?

@Yohahaha, this pr depends on a velox pr: https://github.com/facebookincubator/velox/pull/9425. I will try to push the review progress.

**Comment by Yohahaha** [comment]
> > Hi, could you share the current status?
> 
> @Yohahaha, this pr depends on a velox pr: [facebookincubator/velox#9425](https://github.com/facebookincubator/velox/pull/9425). I will try to push the review progress.

why not choose below usage? I think velox has already support register with prefix, the only thing we need to do is choose which prefixed-function is gluten needed. correct me if it's wrong, thank you!

```
  velox::functions::prestosql::registerAllScalarFunctions("presto_");
  velox::functions::sparksql::registerFunctions("spark_");

  prestoMap("array_distinct" -> "presto");
  // do naming conversion in cpp module
```

**Comment by PHILO-HE** [comment]
> > > Hi, could you share the current status?
> > 
> > 
> > @Yohahaha, this pr depends on a velox pr: [facebookincubator/velox#9425](https://github.com/facebookincubator/velox/pull/9425). I will try to push the review progress.
> 
> why not choose below usage? I think velox has already support register with prefix, the only thing we need to do is choose which prefixed-function is gluten needed. correct me if it's wrong, thank you!
> 
> ```
>   velox::functions::prestosql::registerAllScalarFunctions("presto_");
>   velox::functions::sparksql::registerFunctions("spark_");
> 
>   prestoMap("array_distinct" -> "presto");
>   // do naming conversion in cpp module
> ```

They are discussed internally. We feel it is more clear by using only one registry and we want to avoid maintaining such a map in Gluten.

File: cpp/velox/operators/functions/RegistrationAllFunctions.cc [hunk 1]
```diff
@@ -67,19 +74,19 @@ void registerFunctionOverwrite() {
...
   velox::functions::registerBinaryIntegral<velox::functions::CheckedDivideFunction>({"check_divide"});
+
+  velox::functions::registerPrestoVectorFunctions();
 }
 } // namespace
 
 void registerAllFunctions() {
   // The registration order matters. Spark sql functions are registered after
   // presto sql functions to overwrite the registration for same named
   // functions.
```
Comment (rui-mo) [inline cpp/velox/operators/functions/RegistrationAllFunctions.cc:89]: nit: please revise the comment to be consistent with the implementation.

File: cpp/velox/substrait/SubstraitParser.cc [hunk 2]
```diff
@@ -391,23 +391,13 @@ std::unordered_map<std::string, std::string> SubstraitParser::substraitVeloxFunc
...
     {"murmur3hash", "hash_with_seed"},
     {"xxhash64", "xxhash64_with_seed"},
     {"modulus", "remainder"},
     {"date_format", "format_datetime"},
     {"collect_set", "set_agg"},
-    {"forall", "all_match"},
-    {"exists", "any_match"},
     {"negative", "unaryminus"},
-    {"get_array_item", "get"},
-    {"arrays_zip", "zip"}};
```
Comment (rui-mo) [inline cpp/velox/substrait/SubstraitParser.cc:410]: Have they been registered in Velox sparksql already?

Comment (PHILO-HE) [reply cpp/velox/substrait/SubstraitParser.cc:410]: Yes, see https://github.com/facebookincubator/velox/pull/10298/files.

File: cpp/velox/operators/functions/RegistrationAllFunctions.cc [hunk 3]
```diff
@@ -67,19 +74,16 @@ void registerFunctionOverwrite() {
   velox::exec::registerFunctionCallToSpecialForm(
       kRowConstructorWithAllNull,
       std::make_unique<RowConstructorWithNullCallToSpecialForm>(kRowConstructorWithAllNull));
-  velox::functions::sparksql::registerBitwiseFunctions("spark_");
   velox::functions::registerBinaryIntegral<velox::functions::CheckedPlusFunction>({"check_add"});
   velox::functions::registerBinaryIntegral<velox::functions::CheckedMinusFunction>({"check_subtract"});
   velox::functions::registerBinaryIntegral<velox::functions::CheckedMultiplyFunction>({"check_multiply"});
   velox::functions::registerBinaryIntegral<velox::functions::CheckedDivideFunction>({"check_divide"});
```
Comment (rui-mo) [inline cpp/velox/operators/functions/RegistrationAllFunctions.cc:80]: It seems some registrations in registerFunctionOverwrite() could be removed, including spark_rand, check_xx. After https://github.com/facebookincubator/velox/commit/372abebc3ded838f2f06544b7c4416db0ec4dab1, we need to adapt to the new names checked_xx.

Comment (PHILO-HE) [reply cpp/velox/operators/functions/RegistrationAllFunctions.cc:80]: Yes, I also noted this. Will create another pr to fix it.

**Review by rui-mo — APPROVED** [review]
Thanks!

**Pull request merged**